src/main.cpp

```diff
@@ -8614,21 +8614,20 @@ void ZeroOutResearcherTotals(std::string cpid)
 
 bool LoadAdminMessages(bool bFullTableScan, std::string& out_errors)
 {
-    int nMaxDepth = nBestHeight;
-    int nMinDepth = fTestNet ? 1 : 164618;
-    nMinDepth = pindexBest->nHeight - (BLOCKS_PER_DAY*30*12);
-    if (nMinDepth < 2) nMinDepth=2;
-    if (!bFullTableScan) nMinDepth = nMaxDepth-6;
-    if (nMaxDepth < nMinDepth) return false;
-    CBlockIndex* pindex = blockFinder.FindByHeight(nMinDepth);
-    // These are memorized consecutively in order from oldest to newest
+    // Find starting block. On full table scan we want to scan 6 months back.
+    // On a shallow scan we can limit to 6 blocks back.
+    CBlockIndex* pindex = bFullTableScan
+            ? blockFinder.FindByTime(pindexBest->nTime - MaxBeaconAge())
+            : blockFinder.FindByHeight(pindexBest->nHeight - 6);
+
+    if(pindex->nHeight < (fTestNet ? 1 : 164618))
+       return true;
 
-    while (pindex->nHeight < nMaxDepth)
+    // These are memorized consecutively in order from oldest to newest
+    for(; pindex && pindex->pnext; pindex = pindex->pnext)
```

Comment:
Is it intentional that the highest block is not scanned? If so comment.

Reply:
It probably isn't. I did it like that to retain the behaviour of the previous implementation.